src/overlay/SurveyManager.cpp

```diff
@@ -863,78 +613,8 @@ SurveyManager::processTimeSlicedTopologyRequest(
 void
 SurveyManager::broadcast(StellarMessage const& msg) const
 {
-    uint32_t minOverlayVersion = 0;
-    switch (msg.type())
-    {
-    case SURVEY_REQUEST:
-    case SURVEY_RESPONSE:
-        // Do nothing. All nodes on the network can understand these messages.
-        break;
-    case TIME_SLICED_SURVEY_START_COLLECTING:
-    case TIME_SLICED_SURVEY_STOP_COLLECTING:
-    case TIME_SLICED_SURVEY_REQUEST:
-    case TIME_SLICED_SURVEY_RESPONSE:
-        // Only send messages to nodes that can understand them.
-        minOverlayVersion = TIME_SLICED_SURVEY_MIN_OVERLAY_PROTOCOL_VERSION;
-        break;
-    default:
-        releaseAssert(false);
-    }
     mApp.getOverlayManager().broadcastMessage(
-        std::make_shared<StellarMessage const>(msg), /*hash*/ std::nullopt,
-        minOverlayVersion);
-}
-
-void
-SurveyManager::populatePeerStats(std::vector<Peer::pointer> const& peers,
-                                 PeerStatList& results,
-                                 VirtualClock::time_point now) const
-{
-    size_t resultSize = std::min<size_t>(peers.size(), results.max_size());
-    results.reserve(resultSize);
-
-    for (size_t i = 0; i < resultSize; ++i)
-    {
-        auto& peer = peers[i];
-        auto& peerMetrics = peer->getPeerMetrics();
-
-        PeerStats stats;
-        stats.id = peer->getPeerID();
-        stats.versionStr = peer->getRemoteVersion();
-        stats.messagesRead = peerMetrics.mMessageRead;
-        stats.messagesWritten = peerMetrics.mMessageWrite;
-        stats.bytesRead = peerMetrics.mByteRead;
-        stats.bytesWritten = peerMetrics.mByteWrite;
-
-        stats.uniqueFloodBytesRecv = peerMetrics.mUniqueFloodBytesRecv;
-        stats.duplicateFloodBytesRecv = peerMetrics.mDuplicateFloodBytesRecv;
-        stats.uniqueFetchBytesRecv = peerMetrics.mUniqueFetchBytesRecv;
-        stats.duplicateFetchBytesRecv = peerMetrics.mDuplicateFetchBytesRecv;
-
-        stats.uniqueFloodMessageRecv = peerMetrics.mUniqueFloodMessageRecv;
-        stats.duplicateFloodMessageRecv =
-            peerMetrics.mDuplicateFloodMessageRecv;
-        stats.uniqueFetchMessageRecv = peerMetrics.mUniqueFetchMessageRecv;
-        stats.duplicateFetchMessageRecv =
-            peerMetrics.mDuplicateFetchMessageRecv;
-
-        stats.secondsConnected =
-            std::chrono::duration_cast<std::chrono::seconds>(
-                now - peerMetrics.mConnectedTime.load())
-                .count();
-
-        results.emplace_back(stats);
-    }
-}
-
-void
-SurveyManager::recordResults(Json::Value& jsonResultList,
-                             PeerStatList const& peerList) const
-{
-    for (auto const& peer : peerList)
-    {
-        jsonResultList.append(peerStatsToJson(peer));
-    }
+        std::make_shared<StellarMessage const>(msg));
```

Comment:
Further opportunity for cleanup: minOverlayVersion can now be removed from broadcastMessage as well as Floodgate.

Reply:
Done